app/assets/javascripts/student_profile/services_list.jsx

```diff
@@ -251,11 +251,19 @@ import _ from 'lodash';
       const isConfirming = (this.state.discontinuingServiceId === service.id);
       const isHovering = (this.state.hoveringDiscontinueServiceId === service.id);
       const isPending = (this.props.discontinueServiceRequests[service.id] === 'pending');
+      var Text = 'Confirm';
+
+      if (isPending) {
+        Text = 'Updating...';
+      } else if (isConfirming) {
+        Text = 'Confirm';
+      } else if (service.estimated_end_date == null) {
+        Text = 'Discontinue';
+      } else {
+        Text = 'Discontinue Early';
+      }
```

Comment:
@alexsoble This is good for a review. Basically I rewrote the ternary operator into if else statements to include the extra option for discontinue. The original ternary operator was already nested. It would harm readability if I make it even more nested. This way makes it more readable.

Reply:
@Jngai OK cool, let me check it out!